client/javaGui/freq-tools/src/main/java/org/dash/freq/model/PostPopulationFrequencies.java

```diff
@@ -0,0 +1,290 @@
+    by the Free Software Foundation; either version 3 of the License, or (at
+    your option) any later version.
+
+    This library is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; with out even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this library;  if not, write to the Free Software Foundation,
+    Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA.
+
+    > http://www.gnu.org/licenses/lgpl.html
+
+*/
+package org.dash.freq.model;
+
+import org.dash.freq.controller.DataChecks;
+import org.dash.freq.controller.HeaderProcessor;
+import org.dash.freq.controller.LicenseType;
+import org.dash.freq.view.AppendText;
+import org.dash.freq.view.PhycusGui;
+
+import static org.dishevelled.compress.Readers.reader;
+
+import java.awt.Color;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.TreeMap;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import io.swagger.client.api.CohortApi;
+import io.swagger.client.ApiClient;
+import io.swagger.client.ApiException;
+import io.swagger.client.api.DefaultApi;
+import io.swagger.client.api.PopulationApi;
+import io.swagger.client.model.CohortData;
+import io.swagger.client.model.CohortRequest;
+import io.swagger.client.model.GenotypeList;
+import io.swagger.client.model.HFCurationRequest;
+import io.swagger.client.model.HFCurationResponse;
+import io.swagger.client.model.HaplotypeFrequency;
+import io.swagger.client.model.HaplotypeFrequencyData;
+import io.swagger.client.model.Label;
+import io.swagger.client.model.LabelData;
+import io.swagger.client.model.License;
+//import io.swagger.client.model.License.TypeOfLicenseEnum;
+import io.swagger.client.model.PopulationData;
+import io.swagger.client.model.PopulationRequest;
+import java.util.HashSet;
+
+
+
+/**
+ * PostPopulationFrequencies
+ *
+ */
+public class PostPopulationFrequencies implements Callable<Integer> 
+{
+
+	private File inputFile;
+	private final String gtRegistry;
+	private final String estEntity;
+	private final URL url;
+	private final Population population = new Population();
+	private List<PopulationData> populations;
+
+	private static final String USAGE = "post-population-frequencies [args]";
+	
+	/**
+	 * Post population frequencies to the frequency curation service
+	 *
+	 * @param inputFile
+	 *            input file
+	 * @param accessId
+	 * @param cohortId
+	 * @throws MalformedURLException
+	 */
+	public PostPopulationFrequencies(String gtRegistry, String estEntity) //File inputFile, 
+			throws MalformedURLException 
+	{
+
+		this.url = new URL("http://localhost:8080");
```

Comment:
I apologize, but what properties file? phycus\src\main\resources\application.properties? That doesn't make sense because the standalone distributable won't be able to access it. I think I'm missing something.

Reply:
Things that can change shouldn't be hardcoded in the code. Since this is a GUI, maybe have a preference/option dialog box where you change the server URL instead of a properties file. You can default to `localhost` but should be changeable without re-compiling the code.

Reply:
Sounds good, I'll add that to the options tab.

Reply:
There is now an Advanced button in the options tab with the ability to change urls and reset it to default. (The default is hardcoded into one variable in PhycusGui because the Preferences library requires a default in case the preference in question hasn't been set yet. This means I will only have to change the default in one location when we go live as opposed to 4.)

Total changes: adding the requisite parts to PhycusGui and changing the URL in PostPopulationFrequencies.java to look at the preference variable (or the default value if not set).

Reply:
Awesome!